src/app/components/sale/Entity.tsx

```diff
@@ -0,0 +1,80 @@
+import { EntityDetails } from "@echoxyz/sonar-core";
+import { saleUUID, sonarConfig } from "@/app/config";
+import { EntityCard } from "../entity/EntityCard";
+
+interface EntityProps {
+  loading: boolean;
+  entity?: EntityDetails;
+  error?: Error;
+  authenticated: boolean;
+  walletAddress?: string;
+}
+
+export function Entity({ loading, entity, error, authenticated, walletAddress }: EntityProps) {
+  if (!walletAddress || !authenticated) {
+    return (
+      <div className="flex flex-col gap-2 bg-yellow-50 border border-yellow-200 rounded-lg p-6 w-full">
+        <p className="text-yellow-800 font-medium">Connection Required</p>
+        <p className="text-yellow-700">Connect your wallet and Sonar account to continue with your purchase.</p>
+      </div>
+    );
+  }
```

Comment:
Seems irrelevant to the display of an entity. I don't think an `Entity` component should take anything other than an Entity type really